### PR TITLE
Tools: add ignore information in ros-ubuntu installation shell

### PR DESCRIPTION
**Pull request by lukezhqin**
#22482

### CODE REVIEW

**Comment by rmackay9** [comment]
Hi @lukezhqin,

Thanks very much for this.  Can you change the prefix from "fix:" to "Tools:"?

**Comment by lukezhqin** [comment]
> Hi @lukezhqin,
> 
> Thanks very much for this. Can you change the prefix from "fix:" to "Tools:"?

I've already change the name and the commit comment also, thanks!

**Comment by rmackay9** [comment]
@srmainwaring @jmachuca77 @snktshrma 

how does this look to you guys?

**Comment by rmackay9** [comment]
I've received some external comments from @jmachuca77 who thinks that the "-i" may not be necessary.  Also he wonders if it is a good idea to ignore errors.  Isn't it important to see errors?

**Comment by lukezhqin** [comment]
> I've received some external comments from @jmachuca77 who thinks that the "-i" may not be necessary. Also he wonders if it is a good idea to ignore errors. Isn't it important to see errors?

Indeed, The option "-r" just ignores the errors and keep installing the required dependencies. However, I just install ROS using this script, but  got error then. I refer to this [LINK](https://answers.ros.org/question/194139/turtlebot-2-indigo-installation-problem-cannot-locate-rosdep-for-turtlebot_stdr-turtlebot_dashboard-and-turtlebot_gazebo/) and solve this problem. Can anyone else found same installation issue?

File: Tools/environment_install/install-ROS-ubuntu.sh [hunk 1]
```diff
@@ -235,7 +235,7 @@ if maybe_prompt_user "Add ardupilot-ws to your home folder [N/y]?" ; then
         git clone https://github.com/ArduPilot/ardupilot_ros.git
         popd
         sudo apt update
-        rosdep install --from-paths src --ignore-src --rosdistro=${ROS_DISTRO} -y
+        rosdep install --from-paths src --ignore-src --rosdistro=${ROS_DISTRO} -y -i -r
```
[the remark this answers is not in the document]
Comment (lukezhqin) [reply Tools/environment_install/install-ROS-ubuntu.sh:238]: Sorry, I would just correct this now.

**Comment by srmainwaring** [comment]
There are still some issues with the script when running on an Ubuntu 20.04 VM.

**Host**: macOS Monterey 12.6, Intel Mac Pro
**System**: Ubuntu Focal 20.04.5 LTS on VMware Fusion 13.0

After:

```bash
Add ardupilot-ws to your home folder [N/y]? y
```

we eventually see:

```bash
ardupilot_ros: Cannot locate rosdep definition for [cartographer_ros]
```

This is because the Noetic distro https://github.com/ros/rosdistro/blob/master/noetic/distribution.yaml does not have an entry for `cartographer_ros`. The Melodic distro https://github.com/ros/rosdistro/blob/master/melodic/distribution.yaml does have an entry at L1166:

```yaml
  cartographer_ros:
    doc:
      type: git
      url: https://github.com/googlecartographer/cartographer_ros.git
      version: 1.0.0
    release:
      packages:
      - cartographer_ros
      - cartographer_ros_msgs
      - cartographer_rviz
      tags:
        release: release/melodic/{package}/{version}
      url: https://github.com/ros-gbp/cartographer_ros-release.git
      version: 1.0.0-1
    status: developed
```

=> ignoring the errors with the `-r` flag will cause issues later on some platforms.

**Comment by snktshrma** [comment]
Hi @srmainwaring! 
I am aware of the cartographer issue and working on it to resolve the build issue!
Will make a pr soon!

**Comment by srmainwaring** [comment]
> am aware of the cartographer issue and working on it to resolve the build issue!

@snktshrma do you plan to edit the rosdep yaml (one way would be to add an entry in /etc/apt/sources.list.d/ros-latest.list pointing to a custom yaml maintained by ardupilot, or add another file), or will you take a different approach. Happy to test now I have an environment set up.

**Comment by Ryanf55** [comment]
The dev team is currently only able to support ROS 2. If anyone wants to contribute a fix for the ROS install that works with ROS 1 melodic, go for it, but adding `-r` does not fix the root cause problem on ROS 1 noetic of the package not being available in ROS distro.

**Pull request closed without merging**